qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SingleBrokerDataDeletionTest.java

```diff
@@ -47,284 +47,247 @@
         .untilAsserted(
-            () ->
-                assertThat(getSegmentsCount(broker))
-                    .describedAs("Expected less segments after a snapshot is taken")
-                    .isLessThan(segmentsCount));
+            () -> {
+              reader.seekToFirstEvent();
+              final long firstPositionPostCompaction = reader.getPosition();
+              assertThat(firstPositionPostCompaction).isGreaterThan(firstPositionPreCompaction);
+            });
   }
 
-  private void fillSegments(final Broker broker, final int segmentCount) {
-    writeSegments(broker, segmentCount);
-
-    await()
-        .untilAsserted(
-            () ->
-                assertThat(ControllableExporter.EXPORTED_RECORDS.get())
-                    .describedAs("Expected all written records to be exported")
-                    .isGreaterThanOrEqualTo(writtenRecords.get()));
+  private void deployDummyProcess() {
+    clusteringRule
+        .getClient()
+        .newDeployCommand()
+        .addWorkflowModel(
+            Bpmn.createExecutableProcess("process").startEvent().done(), "process.bpmn")
+        .send()
+        .join();
   }
 
-  private void writeSegments(final Broker broker, final int segmentCount) {
-    while (getSegmentsCount(broker) <= segmentCount) {
-      writeToLog();
-      writtenRecords.incrementAndGet();
-    }
+  private void configureBroker(final BrokerCfg brokerCfg) {
+    final DataCfg data = brokerCfg.getData();
+    data.setSnapshotPeriod(SNAPSHOT_PERIOD);
+    data.setLogSegmentSize(LOG_SEGMENT_SIZE);
+    data.setLogIndexDensity(5);
+    brokerCfg.getNetwork().setMaxMessageSize(MAX_MESSAGE_SIZE);
+
+    final ExporterCfg exporterCfg = new ExporterCfg();
+    exporterCfg.setClassName(ControllableExporter.class.getName());
+    brokerCfg.setExporters(Collections.singletonMap("snapshot-test-exporter", exporterCfg));
   }
 
-  private void writeToLog() {
+  private void publishEnoughMessagesForCompaction() {
+    final int requiredMessageCount = (int) LOG_SEGMENT_SIZE.toBytes() / LARGE_VARIABLE_SIZE;
+    IntStream.range(0, requiredMessageCount + 1).forEach(this::publishMaxMessageSizeMessage);
+  }
 
+  private void publishMaxMessageSizeMessage(final int key) {
     clusteringRule
         .getClient()
         .newPublishMessageCommand()
         .messageName("msg")
-        .correlationKey("key")
+        .correlationKey("msg-" + key)
+        .variables(Map.of("foo", MAX_MESSAGE_SIZE_VARIABLE))
         .send()
         .join();
   }
 
-  private int getSegmentsCount(final Broker broker) {
-    return getSegments(broker).size();
+  @SuppressWarnings("rawtypes")
+  private Stream<Record> newRecordStream(final LogStreamReader reader) {
+    final Spliterator<LoggedEvent> spliterator =
+        Spliterators.spliteratorUnknownSize(reader, Spliterator.ORDERED);
+    return StreamSupport.stream(spliterator, false)
+        .map(event -> CopiedRecords.createCopiedRecord(1, event));
   }
```

Comment:
Fix the `rawtypes` warning.

```suggestion
  private Stream<Record<?>> newRecordStream(final LogStreamReader reader) {
    final Spliterator<LoggedEvent> spliterator =
        Spliterators.spliteratorUnknownSize(reader, Spliterator.ORDERED);
    return StreamSupport.stream(spliterator, false)
        .map(event -> (CopiedRecord<?>) CopiedRecords.createCopiedRecord(1, event));
  }
```

Reply:
Did it make sense why I'm using `CopiedRecord` here? Wasn't sure if it would be obvious - so that we can use `RecordAssert` and get nicer error messages mostly. I was worried it would seem over the top for someone reading it later, but the errors are really much nicer.

Reply:
Make sense :+1: